controller/controller_test.go

```diff
@@ -0,0 +1,139 @@
+package controller
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/rayl17/academy-go-q42021/mocks"
+	"github.com/rayl17/academy-go-q42021/model"
+	"github.com/rayl17/academy-go-q42021/usecase"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPokemonController_GetPokemonByName(t *testing.T) {
+	tests := []struct {
+		name        string
+		pokemonName string
+		want        *model.Pokemon
+		expectedErr error
+	}{
+		{
+			name:        "prueba",
+			pokemonName: "bulbasor",
+			want: &model.Pokemon{
+				Id:         "",
+				Name:       "bulbasor",
+				Type1:      "",
+				Type2:      "",
+				Total:      "",
+				HP:         "",
+				Attack:     "",
+				Defense:    "",
+				SpAttack:   "",
+				SpDefense:  "",
+				Speed:      "",
+				Generation: "",
+				Legendary:  "",
+			},
+			expectedErr: nil,
+		},
+		{
+			name:        "pokemon not found ",
+			pokemonName: "bulbassadivuh",
+			expectedErr: errors.New("Pokemon not found"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &mocks.ServiceInterface{}
```

Comment:
Why you are mocking the service interface? You are testing the controller, you should mock the controller interface.

This is basically the same test on service and usecase.

Remmeber, each layer has it's own responsibility, you should test the logic associated for that responsibility.